BraveRewardsUI/Wallet/Notifications/RewardsNotificationViewBuilder.swift

```diff
@@ -0,0 +1,115 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import BraveRewards
+
+struct RewardsNotificationViewBuilder {
+  
+  static func get(notification: RewardsNotification) -> WalletNotificationView? {
+    switch notification.kind {
+    case .autoContribute, .tipsProcessed, .verifiedPublisher, .insufficientFunds, .pendingNotEnoughFunds:
+      return RewardsNotificationViewBuilder.get(messageNotification: notification)
+    case .grant, .grantAds:
+      return RewardsNotificationViewBuilder.get(actionNotification: notification)
+    default:
+      return nil
+    }
+  }
+  
+  private static func get(actionNotification: RewardsNotification) -> WalletNotificationView? {
+    let category: WalletActionNotification.Category
+    let body: String
+    
+    switch actionNotification.kind {
+    case .grant:
+      body = Strings.NotificationGrantNotification
+      category = .grant
+    case .grantAds:
+      body = Strings.NotificationEarningsClaimDefault
+      category = .grant
+    default:
+      assertionFailure("Undefined case for action notification")
+      return nil
+    }
+    let date = Date(timeIntervalSince1970: actionNotification.dateAdded)
+    return WalletActionNotificationView(
+      notification: WalletActionNotification(
+        category: category,
+        body: body,
+        date: date))
+  }
+  
+  private static func get(messageNotification: RewardsNotification) -> WalletNotificationView? {
+    var category: WalletMessageNotification.Category!
+    let body: String
+    var alertType: WalletAlertNotification.Category?
```

Comment:
You've split these out into multiple types and yet in order to get an alert notification you have to call `get(messageNotification: RewardsNotification)`? Isn't there a better way to set this up so there's only 1 type. Is the message + alert notification the same other than the image?

Reply:
Auto-Contribute notifications can either have a message notification or an alert notification, depending on `BATResult`. Message notification and alert notification have different UI.